tools/static-code-analysis/pmd/suppressions.properties

```diff
@@ -3,3 +3,8 @@ org.eclipse.smarthome.io.console.karaf.internal.OSGiConsole=SystemPrintln
 org.eclipse.smarthome.io.console.rfc147.internal.CommandWrapper=SystemPrintln
 org.eclipse.smarthome.io.console.rfc147.internal.OSGiConsole=SystemPrintln
 org.eclipse.smarthome.core.internal.events.ThreadedEventHandler=EmptyIfStmt
+org.eclipse.smarthome.config.xml.ConfigDescriptionConverter=AvoidCatchingNPE
```

Comment:
This one is rather easy to fix.
The URI constructor will throw a NPE if the provided text is null only (AFAIK).

Reply:
Shouldn't this already do the job?

```diff
@@ -62,12 +62,15 @@ public class ConfigDescriptionConverter extends GenericUnmarshaller<ConfigDescri
             // the URI could be overridden by a context field if it could be
             // automatically extracted
             uriText = (String) context.get("config-description.uri");
+            if (uriText == null) {
+                throw new ConversionException("Missing URI.");
+            }
         }
 
         URI uri = null;
         try {
             uri = new URI(uriText);
-        } catch (NullPointerException | URISyntaxException ex) {
+        } catch (URISyntaxException ex) {
             throw new ConversionException(
                     "The URI '" + uriText + "' in node '" + reader.getNodeName() + "' is invalid!", ex);
         }
```

Reply:
Yes, I updated the PR with the required changes.